### PR TITLE
add the option to restore the engine from backup file

### PR DESCRIPTION
**Pull request by kobihk**

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically. ( keep in mind this list might be overwritten if the job XML is refreshed, for permanent whitelisting, please follow #3 option )
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

**Comment by kobihk** [comment]
ci test please

**Comment by kobihk** [comment]
I verified it with regular deploy and with restore from file deploy

File: tasks/restore_engine_from_file.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+- name: Run engine cleanup command
+  command: "engine-cleanup"
+  when: ovirt_engine_setup_restore_engine_cleanup | bool
```
Comment (mnecas) [inline tasks/restore_engine_from_file.yml:4]: The filter `bool` is needed if it is a string you can remove it.

File: tasks/restore_engine_from_file.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+- name: Add restore file and restore options
+  set_fact:
+    restore_cmd: "{{ restore_cmd }} --{{ item.key }}{% if item.value %}={{ item.value }}{% endif %}"
+  with_dict:
+    - file: "{{ ovirt_engine_setup_restore_file }}"
+    - "{{ ovirt_engine_setup_restore_options | default({}) }}"
+
+- name: Print the restore command
+  debug:
+    var: restore_cmd
```
Comment (mnecas) [inline tasks/restore_engine_from_file.yml:20]: Please remove debug msg.

File: README.md [hunk 3]
```diff
@@ -51,6 +51,10 @@ to ``ovirt_engine_setup_answer_file_path`` variable.
 | ovirt_engine_setup_perform_upgrade    | False                 | If `True`, this role is used to perform an upgrade. |
 | ovirt_engine_setup_product_type       | oVirt                 | One of ["oVirt", "RHV"], case insensitive. |
 | ovirt_engine_setup_offline            | False                 | If `True`, updates for all packages will be disabled. |
+| ovirt_engine_setup_restore_engine_cleanup | False             | Remove the configuration files and clean the database associated with the Engine, relevant only when `ovirt_engine_setup_restore_file` is defined |
+| ovirt_engine_setup_restore_file       | UNDEF                 | Restored the engine with a backup file which created with engine-backup. |
+| ovirt_engine_setup_restore_scopes     | UNDEF                 | List of scopes following values are available: `all`, `files`, `db`, `dwhdb`, `cinderlibdb`. |
```
Comment (mnecas) [inline README.md:56]: just nitpick add square brackets around all options so it is like `ovirt_engine_setup_product_type`

File: defaults/main.yml [hunk 4]
```diff
@@ -32,3 +32,7 @@ ovirt_engine_setup_perform_upgrade: false
 
 ovirt_engine_apache_config_ssl: true
 ovirt_engine_apache_config_root_redirection: true
+
+restore_cmd: 'engine-backup --mode=restore'
```
Comment (mnecas) [inline defaults/main.yml:36]: maybe create `vars` folder with `main.yml` we don't want the user to change the restore_cmd

File: README.md (hunk 3 above)
Comment (kobihk) [reply README.md]: Done

File: defaults/main.yml (hunk 4 above)
Comment (kobihk) [reply defaults/main.yml]: created and removed from defaults

File: tasks/restore_engine_from_file.yml (hunk 1 above)
Comment (kobihk) [reply tasks/restore_engine_from_file.yml:4]: ovirt_engine_setup_restore_engine_cleanup is bool which defined in the defaults file as false

File: tasks/restore_engine_from_file.yml (hunk 2 above)
Comment (kobihk) [reply tasks/restore_engine_from_file.yml]: done

File: tasks/restore_engine_from_file.yml (hunk 1 above)
Comment (mnecas) [reply tasks/restore_engine_from_file.yml:4]: yes but the filter usage is to change `"True"` -> `True` or `"False"` -> `False` etc.
https://docs.ansible.com/ansible/latest/user_guide/playbooks_filters.html

Comment (kobihk) [reply tasks/restore_engine_from_file.yml]: Done

**Pull request merged**